src/Compilers/CSharp/Portable/Symbols/ReferenceManager.cs

```diff
@@ -986,18 +984,13 @@ internal MetadataImportOptions EffectiveImportOptions
 
                 protected override void AddAvailableSymbols(List<AssemblySymbol> assemblies)
                 {
-                    // accessing cached symbols requires a lock
-                    lock (SymbolCacheAndReferenceManagerStateGuard)
+                    CachedSymbols.ForEach<PEAssemblySymbol>(peAssembly =>
                     {
-                        foreach (var assembly in CachedSymbols)
+                        if (IsMatchingAssembly(peAssembly))
```

Comment:
>IsMatchingAssembly [](start = 28, length = 18)

Capturing here and with `assemblies` below.

Reply:
Agree but it's allocation neutral before and after my change: the `foreach` used to allocate. Can do this though to make it better.